scipy/stats/_stats_py.py

```diff
@@ -1005,15 +1018,16 @@ def _moment_outputs(kwds, default_order=1):
 def _moment_result_object(*args):
     if len(args) == 1:
         return args[0]
-    return np.asarray(args)
+    xp = array_namespace(*args)
+    return xp.stack(args)
 
 
 # When `order` is array-like with size > 1, moment produces an *array*
 # rather than a tuple, but the zeroth dimension is to be treated like
 # separate outputs. It is important to make the distinction between
 # separate outputs when adding the reduced axes back (`keepdims=True`).
 def _moment_tuple(x, n_out):
-    return tuple(x) if n_out > 1 else (x,)
+    return tuple(x[i, ...] for i in range(x.shape[0])) if n_out > 1 else (x,)
```

Comment:
See code comment. For better or for worse, this is supposed to iterate over the zeroth axis.